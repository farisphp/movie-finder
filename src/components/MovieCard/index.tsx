import classnames from 'classnames'

import {
  Card,
  CardTitle,
  CardDescription,
  CardContent,
  CardFooter
} from 'ui/card'

interface IMovieCard extends React.HTMLAttributes<HTMLDivElement> {
  image?: string
  title?: string
  description?: string
  isLoading?: boolean
}

/* Will render skeleton loading when isLoading is true */
function MovieCard({
  className,
  image,
  title,
  description,
  isLoading = false
}: IMovieCard) {
  return (
    <Card className={classnames({ 'animate-pulse': isLoading }, className)}>
      <CardContent>
        {isLoading ? (
          <div className="h-[400px] w-full bg-gray-700" />
        ) : (
          <img
            className="h-[400px] w-full"
            style={{ objectFit: 'cover' }}
            src={image}
            alt={title}
          />
        )}
      </CardContent>
      <CardFooter>
        <CardTitle>{title}</CardTitle>
        <CardDescription>{description}</CardDescription>
      </CardFooter>
    </Card>
  )
}

export default MovieCard
