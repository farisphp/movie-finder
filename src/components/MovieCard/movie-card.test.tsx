import { render, screen } from '@testing-library/react'

import MovieCard from '.'

const movieProps = {
  className: 'col-span-12 w-full md:col-span-6 lg:col-span-3',
  image: 'https://image.tmdb.org/t/p/w500//x31PQlhtamAz0FP2vprXImB5Yyn.jpg',
  title: 'The Movie Out Here',
  description:
    'Adam (Robin Nielsen) works as a junior associate for a large Toronto law firm. When a business trip takes him to his hometown of Fernie, he reunites with his old friends, Jason (Viv Leacock) and Theo (James Wallis). Adam gets more than he bargained for when he finds out that, due to a series of terrible business decisions, Theo\'s ski waxing business "Waxopolis" is in serious financial trouble and at risk of being taken over by an evil developer. The three guys hatch a plan to throw a massive fundraising party to try and save the business. But, with the developer and his son on their back, party permit hassles, and a Sasquatch on the loose, will they be able to pull it off?',
  isLoading: false
}

const skeletonProps = {
  className: 'col-span-12 w-full md:col-span-6 lg:col-span-3',
  isLoading: true
}

describe('<MovieCard />', () => {
  it('should render the skeleton loading of movie card', () => {
    const { container } = render(<MovieCard {...skeletonProps} />)
    const movieCard = container.getElementsByClassName('animate-pulse')

    expect(movieCard.item(0)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the movie detail correctly', () => {
    const { container } = render(<MovieCard {...movieProps} />)
    const description = container.getElementsByTagName('p')
    expect(screen.getByRole('img')).toHaveAttribute('src', movieProps.image)

    expect(
      screen.getByRole('heading', { name: movieProps.title, level: 3 })
    ).toBeInTheDocument()

    expect(description.item(0)).toHaveTextContent(movieProps.description)

    expect(container.firstChild).toMatchSnapshot()
  })
})
