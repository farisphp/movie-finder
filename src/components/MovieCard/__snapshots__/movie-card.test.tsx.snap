// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MovieCard /> > should render the movie detail correctly 1`] = `
<div
  class="rounded-lg border bg-card text-card-foreground shadow-sm col-span-12 w-full md:col-span-6 lg:col-span-3"
>
  <div
    class="pb-6"
  >
    <img
      alt="The Movie Out Here"
      class="h-[400px] w-full"
      src="https://image.tmdb.org/t/p/w500//x31PQlhtamAz0FP2vprXImB5Yyn.jpg"
      style="object-fit: cover;"
    />
  </div>
  <div
    class="space-y-2 p-6 pt-0"
  >
    <h3
      class="text-lg font-semibold leading-normal tracking-tight"
    >
      The Movie Out Here
    </h3>
    <p
      class="text-sm text-muted-foreground"
    >
      Adam (Robin Nielsen) works as a junior associate for a large Toronto law firm. When a business trip takes him to his hometown of Fernie, he reunites with his old friends, Jason (Viv Leacock) and Theo (James Wallis). Adam gets more than he bargained for when he finds out that, due to a series of terrible business decisions, Theo's ski waxing business "Waxopolis" is in serious financial trouble and at risk of being taken over by an evil developer. The three guys hatch a plan to throw a massive fundraising party to try and save the business. But, with the developer and his son on their back, party permit hassles, and a Sasquatch on the loose, will they be able to pull it off?
    </p>
  </div>
</div>
`;

exports[`<MovieCard /> > should render the skeleton loading of movie card 1`] = `
<div
  class="rounded-lg border bg-card text-card-foreground shadow-sm animate-pulse col-span-12 w-full md:col-span-6 lg:col-span-3"
>
  <div
    class="pb-6"
  >
    <div
      class="h-[400px] w-full bg-gray-700"
    />
  </div>
  <div
    class="space-y-2 p-6 pt-0"
  >
    <h3
      class="text-lg font-semibold leading-normal tracking-tight"
    />
    <p
      class="text-sm text-muted-foreground"
    />
  </div>
</div>
`;
